src/Event/InvalidateRequestEvent.php

```diff
@@ -2,72 +2,72 @@
 
 namespace Gregurco\Bundle\GuzzleBundleCachePlugin\Event;
 
-use GuzzleHttp\Client;
-use GuzzleHttp\Psr7\UriResolver;
-use Psr\Http\Message\RequestInterface;
 use Psr\Http\Message\UriInterface;
 use Symfony\Component\EventDispatcher\Event;
-use function GuzzleHttp\Psr7\uri_for;
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7;
+use GuzzleHttp\Psr7\Request;
 
 class InvalidateRequestEvent extends Event
 {
-    /** @var RequestInterface[] */
-    private $requests;
-
     /** @var Client */
-    private $client;
+    protected $client;
+
+    /** @var string */
+    protected $method;
+
+    /** @var string */
+    protected $uri;
 
     /**
      * @param Client $client
-     * @param RequestInterface[] $requests
+     * @param string $method
+     * @param string $uri
      */
-    public function __construct(Client $client, array $requests)
+    public function __construct(Client $client, string $method, string $uri)
     {
         $this->client = $client;
-
-        array_walk($requests, [$this, 'addRequest']);
+        $this->method = $method;
+        $this->uri = $uri;
     }
 
     /**
-     * @param RequestInterface $request
+     * @return Client
      */
-    public function addRequest(RequestInterface $request)
+    public function getClient(): Client
     {
-        $request          = $request->withUri($this->buildUri($this->client, $request->getUri()));
-        $this->requests[] = $request;
+        return $this->client;
     }
 
     /**
-     * @param Client $client
-     * @param string $uri
-     *
-     * @return UriInterface
+     * @return string
      */
-    private function buildUri(Client $client, string $uri): UriInterface
+    public function getMethod(): string
     {
-        $uri = uri_for($uri);
-
-        $baseUri = $client->getConfig('base_uri');
-        if (null !== $baseUri) {
-            $uri = UriResolver::resolve(uri_for($baseUri), $uri);
-        }
-
-        return $uri->getScheme() === '' && $uri->getHost() !== '' ? $uri->withScheme('http') : $uri;
+        return $this->method;
     }
 
     /**
-     * @return Client
+     * @return string
      */
-    public function getClient(): Client
+    public function getUri(): string
     {
-        return $this->client;
+        return $this->uri;
     }
 
     /**
-     * @return RequestInterface[]
```

Comment:
One last question for me. Is there a reason to not allow multiple requests ?

Reply:
Yep, I prefer to have one event per one request. I plan to do PR with "delete all cached records from storage" in middleware repo and after that will implement here second event with this functionality. Finally we will have 2 events: delete one request or delete all...

Reply:
ok good idea. maybe name it `purge` instead of `deleteAll`